main.go

```diff
@@ -287,17 +290,31 @@ func apply(configFile string) (ctrl.Options, config.Configuration) {
 	}
 	if err != nil {
 		setupLog.Error(err, "unable to load the config")
-		os.Exit(1)
+		return options, cfg, err
+	}
+
+	if cfg.Integrations != nil {
+		var invalidFrameworks []string
+		for _, framework := range cfg.Integrations.Frameworks {
+			if _, found := jobframework.GetIntegration(framework); !found {
+				invalidFrameworks = append(invalidFrameworks, framework)
+			}
+		}
+		if len(invalidFrameworks) > 0 {
+			err := fmt.Errorf("invalid integration framework(s) %v", invalidFrameworks)
```

Comment:
nit: use `field.NotSupported`. Maybe you can produce an error for each invalid value and aggregate them using `field.ErrorList`